Yosemite/Yosemite/Actions/CardPresentPaymentAction.swift

```diff
@@ -41,10 +41,18 @@ public enum CardPresentPaymentAction: Action {
                         onCardReaderMessage: (CardReaderEvent) -> Void,
                         onCompletion: (Result<PaymentIntent, Error>) -> Void )
 
+    /// Check whether there is a software update available.
     case checkForCardReaderUpdate(onData: (Result<CardReaderSoftwareUpdate, Error>) -> Void,
                         onCompletion: () -> Void)
 
+    /// Update card reader firmware.
     case startCardReaderUpdate(onProgress: (Float) -> Void,
                         onCompletion: (Result<Void, Error>) -> Void)
 
+    /// Restarts the card present payments system
+    /// This might imply, but not be limited to:
+    /// 1. Disconnect from a connected reader
+    /// 2. Clear all credentials, cached data
+    /// 3. Reset all status indicators
+    case restart
```

Comment:
Maybe `reset` would be a better term since this isn't calling `start` again, and if I understand the architecture correctly, the store handling this will be deinitialized and replaced with a new one when we re-authenticate